samza-log4j2/src/main/java/org/apache/samza/logging/log4j2/StreamAppender.java

```diff
@@ -416,7 +416,7 @@ private void startTransferThread() {
    * Helper method to send a serialized log-event to the systemProducer, and increment respective methods.
    * @param logQueueEntry the serialized log-event to be sent to the systemProducer
    */
-  private void sendEventToSystemProducer(EncodedLogEvent logQueueEntry) {
+  protected void sendEventToSystemProducer(EncodedLogEvent logQueueEntry) {
```

Comment:
Same here. This is in PR#1627. Let's keep this PR only on JDK11 runtime support.